expression/constant_propagation.go

```diff
@@ -158,6 +146,51 @@ func (s *propagateConstantSolver) validPropagateCond(cond Expression, funNameMap
 	return nil, nil
 }
 
+// tryToReplaceCond aims to replace all occurances of column 'src' and try to replace it with 'tgt' in 'cond'
+// It returns
+//  bool: if a replacement happened
+//  bool: if 'cond' contains non-deterministic expression
+//  Expression: the replaced expression, or original 'cond' if the replacement is not happened
+func (s *propagateConstantSolver) tryToReplaceCond(src *Column, tgt *Column, cond Expression) (bool, bool, Expression) {
+	if sf, ok := cond.(*ScalarFunction); ok {
+		replaced := false
+		var args []Expression
+		if _, ok := unFoldableFunctions[sf.FuncName.L]; ok {
+			return false, true, cond
+		}
+		// Equality is handled in propagateEQ already
```

Comment:
1. propagateEQ only handles equal functions like `column = constant` but not handle the case like
`a = b and b = funcname()`, should we consider it here?
2. Or may be we do not need propagateEQ now?

Reply:
I've updated the code to support `b = func()` case, thanks